src/metabase/events/notification.clj

```diff
@@ -52,11 +57,29 @@
   (when (supported-topics topic)
     (models.notification/notifications-for-event topic)))
 
+(defn- app-name-trs
+  "Return the user configured application name, or Metabase translated
+  via trs if a name isn't configured."
+  []
+  (or (public-settings/application-name)
+      (trs "Metabase")))
+
+(defn- site-name
+  []
+  (or (public-settings/site-name) (trs "Unknown")))
+
 (defn- enriched-event-info
   [topic event-info]
   ;; DO NOT delete or rename these fields, they are used in the notification templates
-  {:settings    {:application-name (public-settings/application-name)
-                 :site-name        (public-settings/site-name)}
+  {:context     {:application-name (public-settings/application-name)
+                 :site-name        (site-name)
+                 :current-user     @api/*current-user*
+                 ;; extra are set of contexts that are specific to a cerntain emails
+                 ;; currently we need it to support i18n purposes, but ideally it should not exists
+                 :extra            {:user-invited-today         (t/format "MMM'&nbsp;'dd,'&nbsp;'yyyy" (t/zoned-date-time))
+                                    :user-invited-email-subject (trs "You''re invited to join {0}''s {1}" (site-name) (app-name-trs))
```

Comment:
Why do we need to repeat the apostrophes? It might be nice to have a unit test for the fully rendered email template.